<launch>

  <arg name="yaml_file" default="$(find cascade_pid_controller)/config/params.yaml" />

  <arg name="odometry_topic"             default="/groundtruth/base_link/odom"/>
  <arg name="pose_reference_topic"       default="/pose_reference_topic"/>
  <arg name="twist_reference_topic"      default="/twist_reference_topic"/> 
  <arg name="trajectory_reference_topic" default="/trajectory_reference_topic"/> 
  <arg name="cmd_topic"                  default="/mavros/setpoint_raw/attitude"/>
  <arg name="state_topic"                default="/catec_control_manager/state"/>

  <!-- Launch your ROS node -->
  <node name="cascade_pid_controller" pkg="cascade_pid_controller" type="cascade_pid_controller_node" output="screen">
    <rosparam command="load" file="$(arg yaml_file)" />
    
    <param name="odometry_topic"              value="$(arg odometry_topic)"/>
    <param name="pose_reference_topic"        value="$(arg pose_reference_topic)"/>
    <param name="twist_reference_topic"       value="$(arg twist_reference_topic)"/>
    <param name="trajectory_reference_topic"  value="$(arg trajectory_reference_topic)"/> 
    <param name="cmd_topic"                   value="$(arg cmd_topic)"/>
    <param name="state_topic"                 value="$(arg state_topic)"/>
  </node>

</launch>